### PR TITLE
Added the Multikey class

### LINKED ISSUE #33: vc:proof scope
Hi,

In the spec there are references to different proofing techniques, but it is unclear for me  what is proofs and how it guarantees that the data is sound.

In example 11  (see https://www.w3.org/TR/vc-data-model/#proofs-signatures) the proof is embedded in the json structure. It is unclear on what representation is the proof build. 

```
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    "https://www.w3.org/2018/credentials/examples/v1"
  ],
  "id": "http://example.gov/credentials/3732",
  "type": ["VerifiableCredential", "UniversityDegreeCredential"],
  "issuer": "https://example.edu",
  "issuanceDate": "2010-01-01T19:23:24Z",
  "credentialSubject": {
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "degree": {
      "type": "BachelorDegree",
      "name": "Bachelor of Science and Arts"
    }
  },
  "proof": {
    "type": "Ed25519Signature2020",
    "created": "2021-11-13T18:19:39Z",
    "verificationMethod": "https://example.edu/issuers/14#key-1",
    "proofPurpose": "assertionMethod",
    "proofValue": "z58DAdFfa9SkqZMVPxAQpic7ndSayn1PzZs6ZjWp1CktyGesjuTSwRdo
                   WhAfGFCF5bppETSTojQCrfFPP2oumHKtz"
  }
}
```

Since the proof is embedded in the json structure, the proof must be based in this example on another representation without the proof included. 
What is it in this example? Is it:
```
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    "https://www.w3.org/2018/credentials/examples/v1"
  ],
  "id": "http://example.gov/credentials/3732",
  "type": ["VerifiableCredential", "UniversityDegreeCredential"],
  "issuer": "https://example.edu",
  "issuanceDate": "2010-01-01T19:23:24Z",
  "credentialSubject": {
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "degree": {
      "type": "BachelorDegree",
      "name": "Bachelor of Science and Arts"
    }
  }
}
``` 
Or is it the RDF respresentation (which serialisation?)

Suppose it is on the RDF reprsentation then the proof would be valid if I reorder the json to

```
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    "https://www.w3.org/2018/credentials/examples/v1"
  ],
  "id": "http://example.gov/credentials/3732",
  
  "credentialSubject": {
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "degree": {
      "type": "BachelorDegree",
      "name": "Bachelor of Science and Arts"
    }
  },
  "type": ["VerifiableCredential", "UniversityDegreeCredential"],
  "issuer": "https://example.edu",
  "issuanceDate": "2010-01-01T19:23:24Z",
}
``` 

Is this also valid for json based representations?

Taking it one step further, suppose I can reorder the json, what would be the effect on adding additional attributes ?

### PR DESCRIPTION
**Pull request by iherman**
This is to fix #33

### CODE REVIEW

File: vocab/security/vocabulary.yml
Comment (dlongley) [inline vocab/security/vocabulary.yml]: Multikey is a bit more general, so I suggest this:

```suggestion
    comment: Verification method to be used with, for example, data integrity proof cryptographic suites, such as the eddsa-2022 cryptographic suite. See the [EdDSA Cryptosuite v2022](https://www.w3.org/TR/vc-di-eddsa/#multikey) specification for further details.
```

**Review by dlongley — APPROVED** [review]

File: vocab/security/vocabulary.yml
Comment (OR13) [inline vocab/security/vocabulary.yml]: Why ?

**Review by OR13 — CHANGES_REQUESTED** [review]
You be good to discuss PRs like this in the group.

File: vocab/security/vocabulary.yml
Comment (iherman) [reply vocab/security/vocabulary.yml]: Why on which of the two statements?

For both cases I did not invent anything, just followed what is in the spec and/or in the other PR-s, that is all.

@msporny @dlongley ?

Comment (msporny) [reply vocab/security/vocabulary.yml]: I think what @OR13 is saying is that `Ed25519Signature2020` is a type of `Proof`? That, or he's asking why we're deprecating the signature type? I can't tell.

But, to be clear,  `Ed25519Signature2020` is specifically not a type of `VerificationMethod` -- verification methods are mechanisms used to determine if a proof is valid.

**Comment by msporny** [comment]
@OR13 wrote:
> You be good to discuss PRs like this in the group.

If you're going to block a PR, please provide a concrete change request in your review. I have requested a re-review from you.

File: vocab/security/vocabulary.yml
Comment (msporny) [inline vocab/security/vocabulary.yml:47]: I don't know why we're labeling this with `eddsa-2022`, since Multikey is a key format, not a specific cryptosuite?

Comment (msporny) [inline vocab/security/vocabulary.yml]:
```suggestion
      - label: EdDSA Cryptosuite v2022
```

I don't know why we're labeling this with eddsa-2022, since Multikey is a key format, not a specific cryptosuite? Maybe the following above?

File: vocab/security/vocabulary.yml
Comment (msporny) [inline vocab/security/vocabulary.yml]:
```suggestion
    upper_value: sec:Proof
```

**Review by msporny — CHANGES_REQUESTED** [review]
+1 to the PR in general. Two changes that will need to be made for it to be merged.

**Review by OR13 — CHANGES_REQUESTED** [review]
I don't understand the PR, it seems the intention is to normatively define a key format for use with securing formats, but it is deprecating properties as well.

I prefer to discuss such changes on a call.

I do not think we should add key representations to a TR document without ever discussing them as a group.

**Comment by msporny** [comment]
> I don't understand the PR, it seems the intention is to normatively define a key format for use with securing formats, but it is deprecating properties as well.

@iherman can you please move the `Ed25519Signature2020` deprecation to another PR and just make this one about Multikey?

> I prefer to discuss such changes on a call.
> I do not think we should add key representations to a TR document without ever discussing them as a group.

The group is aware of Multikey's existence and the usage thereof; we have multiple implementers that have demonstrated interop on it (via `did:key`) and multiple implementers having written into the WG about implementing it and supporting it: 

https://lists.w3.org/Archives/Public/public-vc-wg/2023Jan/0027.html

That said, feel free to mention it on the calls this week and ask for input from people on this PR.

**Comment by iherman** [comment]
> > I don't understand the PR, it seems the intention is to normatively define a key format for use with securing formats, but it is deprecating properties as well.
> 
> @iherman can you please move the `Ed25519Signature2020` deprecation to another PR and just make this one about Multikey?

I rolled back the `Ed25519Signature2020` change.

**Comment by iherman** [comment]
> I prefer to discuss such changes on a call.
> I do not think we should add key representations to a TR document without ever discussing them as a group.

This PR (and any change on the vocabulary) just follows the standard. The current spec introduces a "Multikey" type in https://www.w3.org/TR/vc-di-eddsa/#multikey as a class which also says 

> type of the verification method _MUST_ be `Multikey`

which means that `Multikey` must be a class (if it can appear as on object in a `type` statement), and the spec talks about a verification method be of that type, so the way I understand it it must be a subclass of `VerificationMethod`

And I have just translated what is in the text into the vocabulary. B.t.w., we should check all the various terms we use in any of the specs the same way.

**Comment by iherman** [comment]
> > > I don't understand the PR, it seems the intention is to normatively define a key format for use with securing formats, but it is deprecating properties as well.
> > 
> > 
> > @iherman can you please move the `Ed25519Signature2020` deprecation to another PR and just make this one about Multikey?
> 
> I rolled back the `Ed25519Signature2020` change.

I have also created #95 for the deprecation.

**Comment by OR13** [comment]
Based on the call today, I am approving this, it seems we are good to make these changes.

**Review by OR13 — APPROVED** [review]
Approving, modulo @msporny feedback being applied.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-04-26-vcwg#section2-3) on 2023-04-26

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.3. Added the Multikey class (pr vc-data-integrity#93)

_See github pull request [vc-data-integrity#93](https://github.com/w3c/vc-data-integrity/pull/93)._




**Manu Sporny:** Next up is data integrity, couple of stuck PRs here.  
… orie, I think you wanted group to be aware of multikey class added to vocab (vc-data-integrity #93).  

**Orie Steele:** my question was, for context, there are references from new vc-di-bbs work item to this, so I'm trying to understand from a normative standpoint, can I reference.  
… multikey from vc-di-bbs?  
… wasn't sure if group has decided to create normative key format.  

> *Kristina Yasuda:* would encourage the WG to review 1101 and 1100 before the next call.

**Manu Sporny:** I think the reference is fine, but agree we need to make a decision as a WG as to where the reference goes.  
… multikey is the key format used in [did:key,](did:key,) which has lots of implementations.  

> *Orie Steele:* Here is the todo, in vc-di-bbs that points to Multikey: [https://w3c.github.io/vc-di-bbs/#multikey](https://w3c.github.io/vc-di-bbs/#multikey).

**Manu Sporny:** eddsa, ecdsa, maybe now bbs.  
… we have been creating normative version of this in the cryptosuites themselves, e.g., here are the key formats this suite supports...  
… I think the proposal right now is for each suite to keep defining sub-parts of multikey spec, and instead of spreading them around like that why don't we put them into the data integrity specification (normatively) so that all suites can point to one place?  

> *Orie Steele:* +1 to everything Manu is saying, it feels like we have made improvements for key formats that align with how we now have DataIntegrityProof once, and we don't define a new type for each new suite, multikey seems in that spirit.

**Manu Sporny:** Can a cryptosuite decide to define its own multikey identifier (probably have to do this in bbs).  

> *Dave Longley:* +1 to just put Multikey definition in the Data Integrity Proof spec as a key format that cryptosuites can use.

> *Orie Steele:* +1 to (keeping) Multikey in data integrity, and citing it from vc-di-bbs (and others).

> *Orie Steele:* +1 to allowing suites to not use Multikey.

> *Dave Longley:* +1 to one place (data integrity spec).

**Manu Sporny:** any objections to putting multikey into data integrity spec?  

> *Phillip Long:* +1 to allow Multikey in the data integrity spec.

> *Chris Abernethy:* (no objections voiced).

**Manu Sporny:** we will do that, and put def into data integrity spec.  
… should cryptosuites be able to define key formats that they support? e.g., we support multi-key, but only "this subset". Would anyone object to this?  

> *Phillip Long:* +1 to allowing cryptosuites to define public keys they support.

> *Dave Longley:* +1 to allow cryptosuites to define key types they support.

> *Orie Steele:* +1 to allow cryptosuites to define key types they support.

> *Chris Abernethy:* (no objections voiced).

**Manu Sporny:** Orie does that address your concerns.  

**Orie Steele:** yes.  
… talking about clarification of `@context`.  

**Kristina Yasuda:** encourage folks to keep reviewing PRs, 1100 and 1101 appear to be crucial.  

---

</details>

**Review by msporny — APPROVED** [review]

**Comment by msporny** [comment]
Substantive, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**
